libs/json-schema-to-zod-v3/src/utils.ts

```
@@ -0,0 +1,126 @@
+ * Safely gets a property from an object using Object.getOwnPropertyDescriptor
+ * This is more robust than simple property access for special properties
+ *
+ * @param obj - Object to get property from
+ * @param prop - Property name
+ * @returns True if property exists on object
+ *
+ * @example
+ * hasOwnProperty({ a: 1 }, 'a') // true
+ * hasOwnProperty({ a: 1 }, 'b') // false
+ */
+export function hasOwnProperty(obj: any, prop: string): boolean {
```

Biome (2.1.2)

[error] Do not shadow the global "hasOwnProperty" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)